### PR TITLE
Add Dataset integration tests - Table Columns

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature: Testing

### Detail
Follow-up of #1391 

- Implement Table Column tests

### Relates
- https://github.com/data-dot-all/dataall/issues/1358
- #1391 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Tested locally:
image

**Comment by dlpzx** [comment]
Testing in real CICD pipeline

File: tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py [hunk 1]
```diff
@@ -7,35 +8,81 @@
...
-    pass
+@pytest.mark.parametrize(
+    'dataset_fixture_name,tables_fixture_name',
+    [('session_s3_dataset1', 'session_s3_dataset1_tables')],
+)
+def test_sync_dataset_table_columns_unauthorized(client2, dataset_fixture_name, tables_fixture_name, request):
+    dataset = request.getfixturevalue(dataset_fixture_name)
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    assert_that(sync_dataset_table_columns).raises(GqlError).when_called_with(client2, table_uri).contains(
```
Comment (SofiaSazonova) [inline tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:36]: Does it actually catcher GqlError? When I test locally (with AWS deployment) the same thing returns just Exception(

Comment (SofiaSazonova) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:36]: Did you test it in your deployment (not locally)?

File: tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py [hunk 2]
```diff
@@ -7,35 +8,81 @@
...
+@pytest.mark.parametrize(*TABLES_FIXTURES_PARAMS)
+def test_update_dataset_table_column(client1, tables_fixture_name, request, session_id):
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    columns = list_dataset_table_columns(client1, table_uri)
+    column_uri = columns.nodes[0].columnUri
+    response = update_dataset_table_column(
+        client1, column_uri, {'description': f'{session_id} new updated description'}
+    )
+    assert_that(response.description).is_equal_to(f'{session_id} new updated description')
```
Comment (SofiaSazonova) [inline tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:73]: nit: I would minimize risk of accidental typos, if we are to change this
```
new_desc = f'{session_id} new updated description'
response = update_dataset_table_column(
        client1, column_uri, {'description': new_desc}
    )
assert_that(response.description).is_equal_to(new_desc)
```

File: tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py (hunk 1 above)
Comment (dlpzx) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:36]: Yes, and the tests for the table_columns pass:

image


There is an issue for the update_persisten_env which I am looking into. It should not be related to this, but I am waiting to merge until I get the issue
image

Comment (dlpzx) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:36]: The error on the teardown of resources might be related to the metadata forms delete:
image
The error in the test of the persistent environment is a Timeout

Comment (SofiaSazonova) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py:36]: Worksheet is not yet connected to MFs. In any way. That's strange

File: tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py (hunk 2 above)
Comment (dlpzx) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py]: Done!

**Review by SofiaSazonova — APPROVED** [review]

**Pull request merged**